cou/apps/auxiliary.py

```diff
@@ -457,3 +465,142 @@ async def _verify_nova_compute(self, target: OpenStackRelease) -> None:
             raise ApplicationError(
                 f"Units '{', '.join(units_not_upgraded)}' did not reach {target}."
             )
+
+
+@AppFactory.register_application(["vault"])
+class Vault(AuxiliaryApplication):
+    """Application for vault."""
+
+    wait_timeout = LONG_IDLE_TIMEOUT
+    wait_for_model = True
+
+    def _check_snap_vault_installed(self) -> None:
+        """Make sure snap vault is installed.
+
+        :raises SnapVaultNotInstalled: When no snap valut installed.
+        """
+        result = subprocess.run(
```

Comment:
we should not limit the user to use snap

Reply:
+1 I agree.  We should only check:

- if we can run the `vault` command
- if so, if `vault` is a compatible version (this can be out of scope here; we can follow up later with version checks)

Reply:
We force because it has better security.

Reply:
Sorry I don't agree; cou should not be reliant on a single distribution of vault.  Some customer clouds may have different requirements and have vault installed via a different method.

It would be a hard requirement for me that cou does not depend on the vault snap.

Reply:
If we want extra security, we can warn the user if vault is found but it's not installed from the snap.  We can also print the full path to the `vault` command we find.

Reply:
You want to search `vault` in user's `$PATH`? I feel it's not precise.

Reply:
+1 to not limit the vault from snap. We can have some default path when we search vault. If it doesn't fit in some case, we can let user assign the path of vault  from the command line parameter

Reply:
Why we are not running this on charm? Will cou snap even access to execute vault?
Sorry I'm maybe out of context here, but this looks as not good idea and I agree that cou should not depend on version of vault installed by user. If it failed for someone, it will be super hard to reproduce, since you have no idea what `vault` was used.
If vault is needed to be used locally, can we pack proper version with our snap?

Reply:
> Why we are not running this on charm?

Because juju log will record the unseal key.

Reply:
I change it to just `vault`, resolved.